src/main/java/com/turtywurty/railroad/fetch/version/SemVer.java

```diff
@@ -0,0 +1,71 @@
+package com.turtywurty.railroad.fetch.version;
+
+public class SemVer implements Comparable<SemVer> {
+
+	private final Integer major;
+	private final Integer minor;
+	private Integer patch;
+
+	public SemVer(String versionString) {
+		String[] vs = versionString.split("\\.");
+		this.major = Integer.parseInt(vs[0]);
+		this.minor = Integer.parseInt(vs[1]);
+
+		if (vs.length == 3)
+			this.patch = Integer.parseInt(vs[2]);
+	}
+
+	public int getMajor() {
+		return major;
+	}
+
+	public int getMinor() {
+		return minor;
+	}
+
+	public int getPatch() {
+		return patch;
+	}
+
+	@Override
+	public int compareTo(SemVer other) {
+		if (this.major > other.major) {
```

Comment:
this can be simplified with Integer.compareTo

Reply:
done.